pkgs/development/python-modules/ismartgate/default.nix

```diff
@@ -1,66 +1,58 @@
 { lib
-, asynctest
 , buildPythonPackage
-, click
-, defusedxml
-, dicttoxml
 , fetchFromGitHub
-, httpx
+, pythonOlder
+  # build inputs
+, dicttoxml
+, defusedxml
 , pycryptodome
-, pytest-asyncio
-, pytest-raises
+, httpx
+  # check inputs
 , pytestCheckHook
-, pythonOlder
-, respx
+, pytest-runner
 , typing-extensions
+, respx
+, pytest-asyncio
+, click # missing from setup.py
 }:
 
 buildPythonPackage rec {
   pname = "ismartgate";
-  version = "4.0.4";
+  version = "5.0.0";
   format = "setuptools";
 
-  disabled = pythonOlder "3.7";
+  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "bdraco";
     repo = pname;
-    rev = "v${version}";
-    hash = "sha256-yh7gPyy3VMdyINBCZo5K2wA0BY7yYgHrKGZRB/pm77U=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-o2yzMxrF0WB6MbeL1Tuf0Sq4wS4FDIWZZx1x2rvwLmY=";
   };
 
   propagatedBuildInputs = [
-    click
-    defusedxml
     dicttoxml
-    httpx
+    defusedxml
     pycryptodome
-    typing-extensions
+    httpx
   ];
 
   nativeCheckInputs = [
-    asynctest
-    pytest-asyncio
-    pytest-raises
     pytestCheckHook
-    respx
   ];
 
-  postPatch = ''
-    substituteInPlace setup.py \
-      --replace '"pytest-runner>=5.2",' ""
-  '';
+  checkInputs = [
+    pytest-runner
+    typing-extensions
```

Comment:
This is a [run-time requirement](https://github.com/bdraco/ismartgate/blob/v5.0.0/ismartgate/cli.py#L11) for the CLI part.